src/resources/Catalogs/ProductListingConfigurationInterfaces.ts

```diff
@@ -1,13 +1,21 @@
 import {QueryFilterModel} from './ProductListingInterfaces';
 
 export interface ProductListingConfigurationModel {
+    /**
+     * The id of the ProductListingConfiguration.
+     */
+    id?: string;
```

Comment:
An ID is optional? :suspect:

Reply:
Well you don't need it when you create a Listing config but it gets return when you fetch one.

Do you think having models for use cases would be better in this situation ?

Reply:
Actually, please edit `ProductListingConfiguration.ts` to specify the return type of `get` and `list`.
If you always get `id` in those two calls, put it non-optional, because every time that `ProductListingConfigurationModel` is used as a parameter, it's used with `New<ProductListingConfigurationModel>` which removes the id.

Reply:
I think I got it. 

Thanks I will definitely reach out to you if what I did wasn't what you had in mind.